ui/resolution_selection.js

```diff
@@ -95,7 +99,22 @@ shaka.ui.ResolutionSelection = class extends shaka.ui.SettingsMenu {
     // If there is a selected variant track, then we filter out any tracks in
     // a different language.  Then we use those remaining tracks to display the
     // available resolutions.
-    const selectedTrack = tracks.find((track) => track.active);
+    let selectedTrack = tracks.find((track) => track.active);
+    if (this.flag ==1 && tracks.find((track) =>
+      track.height ==
+      Number((this.currentSelection.textContent.toString()).slice(0, -1))) !==
+      undefined) {
+      selectedTrack = tracks.find((track) => track.height ==
+      Number((this.currentSelection.textContent.toString()).slice(0, -1)));
+      const tempTrack = tracks.find((track) => track.active);
+      for (let i=0; i < tracks.length; i++) {
+        if (tracks[i].id === tempTrack.id) {
+          tracks[i].active = false;
```

Comment:
Setting active does nothing as the objects returned are created each call.  You need to use `player.selectVariantTrack`.  See also `onTrackSelected_`.